EIPS/eip-2938.md

```diff
@@ -0,0 +1,259 @@
+
+Transaction validity is defined rigidly by the protocol, forcing a particular set of validity rules based on ECDSA signatures and nonces onto the protocol. This precludes innovation in a number of important areas, particularly including:
+
+1. Smart contract wallets that use signature verification other than ECDSA (eg. Schnorr, BLS, post-quantum...)
+2. Smart contract wallets that include features such as multisig verification or social recovery, reducing the highly prevalent risk of funds being lost or stolen
+3. Privacy-preserving systems like [tornado.cash](http://tornado.cash)
+4. Attempts to improve gas efficiency of defi protocols by preventing transactions that don't satisfy high-level conditions (eg. existence of a matching order) from being included on chain
+5. Users being able to pay for transaction fees in a token other than ETH (eg. by converting that token into the ETH needed for fees inside the transaction in real-time)
+
+In most of the above cases, it is still possible to do these applications using intermediaries, most notably the [Gas Station Network](https://www.opengsn.org/) and application-specific alternatives, but these alternatives are (i) technically inefficient, due to the extra 21000 gas to pay for the relayer, (ii) economically inefficient, as relayers need to make a profit on top of the gas fees that they pay. Additionally, use of intermediary protocols means that these applications cannot simply rely on base ethereum infrastructure and need to rely on extra protocols that have smaller userbases and higher risk of no longer being available at some future date.
+
+Account abstraction allows for the validity conditions of a transaction to include execution of arbitrary EVM code (with some limits regarding what state can be accessed), allowing the above use cases.
+
+We split account abstraction into two tiers: **single-tenant AA**, which is intended to support abstracted accounts that represent wallets or otherwise are only meant to be accessed by one or a few participants, and **multi-tenant AA**, which is intended to allow abstracted accounts to also directly be applications (eg. tornado.cash, Uniswap). Out of the four use cases above, it can be approximately said that single-tenant AA supports (1) and (2), and multi-tenant AA supports (3) and (4). We get into the differences between the two tiers in the specification and rationale section below.
+
+## Specification (Single-tenant AA)
+
+After `FORK_BLOCK`, the following changes will be recognized by the protocol.
+
+### Constants
+
+| Constant | Value |
+| - | - |
+| **`AA_ENTRY_POINT`** | `0xffffffffffffffffffffffffffffffffffffffff` |
+| **`AA_TX_TYPE`** | `2` |
+| **`FORK_BLOCK`** | TBD |
+| **`AA_BASE_GAS_COST`** | 15000 |
+
+### New Transaction Type
+
+A new EIP-2718 transaction with type `AA_TX_TYPE` is introduced. Transactions of this type are referred to as "AA transactions". Their payload should be interpreted as `rlp([nonce, target, data])`.
```

Comment:
Can the "EIP-2718" mention be supplemented by the EIP title (in brackets) here for readability?